.github/workflows/spm-update.yml

```diff
@@ -0,0 +1,103 @@
+name: Swift Package Update
+
+on:
+  schedule:
+    - cron:  '0 0 * * 0'
+  workflow_dispatch:
+  
+jobs:
+  macos:
+    name: macOS ${{ matrix.configuration }} test build
+    runs-on: macos-latest
+    strategy:
+      matrix:
+        configuration: [debug, release]
+    steps:
+    - uses: actions/checkout@v2
+    - uses: maxim-lobanov/setup-xcode@v1.1
+      with:
+        xcode-version: latest
+    - uses: actions/cache@v2
+      with:
+        path: .build
+        key: ${{ runner.os }}-spm-cache-${{ hashFiles('Package.resolved') }}
+    - name: Check Xcode version
+      run: xcodebuild -version
+    - name: Check Swift version
+      run: swift --version
+    - name: Update Swift Packages
+      run: swift package update
+    - name: Build Release
+      if: matrix.configuration == 'release'
+      run: swift build --configuration ${{ matrix.configuration }}
+    - name: Build $ Test
+      if: matrix.configuration == 'debug'
+      run: swift test --configuration ${{ matrix.configuration }}
+  linux:
+    name: Linux ${{ matrix.linux }} ${{ matrix.configuration }} test build
+    container:
+      image: swift:${{ matrix.linux }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        linux: [bionic, xenial, focal, amazonlinux2, centos8]
+        configuration: [debug, release]
+    steps:
+    - uses: actions/checkout@v2
+    - name: Install libsqlite3
+      if: ${{ !(startsWith( matrix.linux, 'centos' ) || startsWith( matrix.linux, 'amazonlinux' )) }}
+      run: apt-get update && apt-get install -y --no-install-recommends libsqlite3-dev
+    - name: Install libsqlite3
+      if: ${{ startsWith( matrix.linux, 'centos' ) || startsWith( matrix.linux, 'amazonlinux' ) }}
+      run: yum update -y && yum install -y sqlite-devel
+    - uses: actions/cache@v2
+      with:
+        path: .build
+        key: ${{ runner.os }}-${{matrix.linux}}-spm-cache-${{ hashFiles('Package.resolved') }}
+    - name: Check Swift version
+      run: swift --version
+    - name: Update Swift Packages
+      run: swift package update
+    - name: Build
+      if: matrix.configuration == 'release'
+      run: swift build --configuration ${{ matrix.configuration }}
+    - name: Build & Test
+      if: matrix.configuration == 'debug'
+      run: swift test --configuration ${{ matrix.configuration }} --enable-test-discovery
+  
```

Comment:
As noted here there seems to be an error with the empty line in the workflow file: https://github.com/Apodini/Apodini/actions/runs/543502065/workflow

Reply:
this run is super old, basically my first commit. I developed the workflow in my own fork: https://github.com/tgymnich/apodini. Feel free to look at the pull requests and past action runs. If you want to try it yourself just fork my fork. Also I omitted `needs: [macos, linux]` in the createPR workflow to speed up testing in my fork. The workflow in this PR waits for the test on macOS and linux to finish.

Reply:
Thanks a lot! I checked out the fork, great to see the workflow working in the fork 👍